rdt-open-reader/src/main/java/io/ona/rdt/sync/RDTSyncIntentService.java

```diff
@@ -42,4 +52,30 @@ private void saveLocationTree() {
         String locationTreeJson = new Gson().toJson(locationTree);
         allSharedPreferences.savePreference(Constants.Preference.LOCATION_TREE, locationTreeJson);
     }
+
+    @Override
+    protected void init(@NonNull Context context) {
+        super.init(context);
+        try {
+            Field syncUtilField = getClass().getSuperclass().getDeclaredField("syncUtils");
+            syncUtilField.setAccessible(true);
+            syncUtilField.set(this, new RDTSyncUtils(getBaseContext()));
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static class RDTSyncUtils extends SyncUtils {
+
+        public RDTSyncUtils(Context context) {
+            super(context);
+        }
+
+        @Override
+        public void logoutUser() throws AuthenticatorException, OperationCanceledException, IOException {
+            if (!CovidLoginActivity.isRunning) {
+                super.logoutUser();
+            }
+        }
+    }
```

Comment:
What is happening here?

Reply:
actually in client-core whenever the logout then it opens the LoginActivity even it's already opened, so to overcome this issue we need to check if LoginActivity is already running then skip the logout, and in client-core, we don't have an option to override this method even the SyncUtils instance create privately.

Reply:
why would logout happen twice?

Reply:
also, it's best to address this as a separate issue. Create a separate issue and describe the problem so we can continue the discussion there. For now I think this should be removed.